lib/os/spsc_pbuf.c

```diff
@@ -213,15 +229,15 @@ void spsc_pbuf_commit(struct spsc_pbuf *pb, uint16_t len)
 	uint32_t wr_idx = *wr_idx_loc;
 
 	sys_put_be16(len, &data_loc[wr_idx]);
-	__sync_synchronize();
+	barrier_isync_fence_full();
```

Comment:
what do you want to achieve here? Because my feeling is that you want to use `barrier_dsync_fence_full` or at least `barrier_dmem_fence_full` here.